tests/integration/CapabilityTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Automattic\LegacyRedirector\Tests;
+
+use Automattic\LegacyRedirector\Capability;
+
+/**
+ * CapabilityTest class
+ */
+class CapabilityTest extends TestCase {
+
+	protected static $initialized = false;
+
+	public function setUp() {
+		if ( ! self::$initialized ) {
+			$capability = new Capability();
+			$capability->register();
+
+			self::$initialized = true;
+		}
```

Comment:
To test the logic in `includes/class-wpcom-legacy-redirector.php` (about whether the Capability method is called or not), then you may want to have this moved to each method instead of in this setUp.

If anything is left in the setUp, then you may need a tearDown as well.

Reply:
I've added the logic, but replaced the setUp with a beforeClass, we could skip this but I want to make sure we have no capabilities setup before the tests.

I use tearDown to check the register init and making sure it's not being setup again after the initial init.